File: abc033/src/b.go
package main

import (
	"fmt"
)

func main() {
    var n int
    fmt.Scan(&n)
    
    sp := make([]struct{ s string; p int64 }, n)
    var sum int64 = 0
    for i := 0; i < n; i++ {
        fmt.Scan(&sp[i].s, &sp[i].p)
        sum += sp[i].p
    }
    
    for _, city := range sp {
        if city.p > sum/2 {
            fmt.Println(city.s)
            return
        }
    }
    fmt.Println("atcoder")
}

File: abc033/src/b.cpp
//correct 
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<pair<string, long long>> sp(n); // define a vector of pairs 
    for (int i = 0; i < n; i++) {
        cin >> sp[i].first >> sp[i].second; // read in the city name and population
    }
    // loop through the vector and add up all the populations
    long long sum = 0;
    for (auto [_, p_i] : sp) { 
        sum += p_i;
    }
    //check the population of city is greater than half of the total population
    for (auto [s_i, p_i] : sp) {
        if (p_i > sum / 2) {
            cout << s_i << endl;
            return 0;
        }
    }

    cout << "atcoder" << endl;

    return 0;
}